api/v1alpha1/composable_webhook.go

```diff
@@ -0,0 +1,271 @@
+				}
+			}
+		default:
+			composablelog.Info("findGetValueFrom", "key", mykey, "type-unknown", vv)
+		}
+	}
+	return allErrs
+}
+
+// findGetValueFrom2 functions in the same way as findGetValueFrom above except taking []interface{} as inputs
+// findGetValueFrom2 processes the ararys in the template content for getValueFrom elements
+func (r *Composable) findGetValueFrom2(path *field.Path, m []interface{}) field.ErrorList {
+	var allErrs field.ErrorList
+	for k, v := range m {
+		mykey := path.Child(strconv.Itoa(k))
+		switch vv := v.(type) {
+		case string, int32, int64, float32, float64, bool:
+		case []interface{}:
+			if err := r.findGetValueFrom2(mykey, vv); err != nil {
+				allErrs = append(allErrs, err...)
+			}
+		case map[string]interface{}:
+			if vv["getValueFrom"] != nil {
+				newM, _ := json.Marshal(vv["getValueFrom"])
+				myGetValueFrom := ComposableGetValueFrom{}
+				json.Unmarshal(newM, &myGetValueFrom)
+				composablelog.Info("found getValueFrom", "key", "getValueFrom", "value", vv["getValueFrom"])
+				if err := validateGetValueFrom(mykey.String(), myGetValueFrom); err != nil {
+					composablelog.Info("getValueFrom is INVALID", "detail", err)
+					allErrs = append(allErrs, field.Invalid(mykey.Child("getValueFrom"), r.Name, err.Error()))
+				}
+				composablelog.Info("getValueFrom is valid")
+				// TODO: set a random value of appropriate type for dry-run
+				m[k] = "abc2"
+			} else {
+				if err := r.findGetValueFrom(mykey, vv); err != nil {
+					allErrs = append(allErrs, err...)
+				}
+			}
+		default:
+			composablelog.Info("findGetValueFrom", "key", k, "type-unknown", vv)
+		}
+	}
+	return allErrs
+}
+
+// validateGetValueFrom validates the syntax of input getValueFrom fields
+func validateGetValueFrom(k string, getValueFrom ComposableGetValueFrom) error {
+	var missingItems []string
+	if getValueFrom.Name == "" {
+		missingItems = append(missingItems, "name")
+	}
+	if getValueFrom.Kind == "" {
+		missingItems = append(missingItems, "kind")
+	}
+	if getValueFrom.Path == "" {
+		missingItems = append(missingItems, "path")
+	}
+	if len(missingItems) > 0 {
+		items := array2string(missingItems)
+		return fmt.Errorf("Missing required field(s) - %v", items)
+	}
+	return nil
+}
+
+func array2string(a []string) string {
+	str := ""
+	for _, v := range a {
+		str = v + "," + str
+	}
+	str = strings.TrimRight(str, ",")
+	return str
+}
+
+// dryrun as a means of syntax validation of the template content
+func (r *Composable) dryRun(m map[string]interface{}, op string) *field.Error {
+	composablelog.Info("dryRun", "name", r.Name)
+	newM, _ := json.Marshal(m)
+	composablelog.Info("embedded request", "new", string(newM))
+
+	cl, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		return field.Invalid(field.NewPath("spec").Child("template"), r.Name, "dry-run failed to get client")
+	}
+
+	u := unstructured.Unstructured{Object: m}
+	u.SetAPIVersion(m["apiVersion"].(string))
```

Comment:
Maybe I'm wrong but the this and the next line look redundant. You create `unstructured` from `m` so if `m` contains `apiVersion` and `kind` the `unstructured will contain them too. Is it correct?
Furthermore, if `m` dosn't contain the keys, the interface conversion `m["apiVersion"].(string)` will produce the following error "panic: interface conversion: interface {} is nil, not string"

Reply:
You are right. Remove them.